### PR TITLE
Feature/constrain available blocks

### PR DESCRIPTION
**Pull request by sbrody**
Limits the blocks available to the editor to those listed here: https://trello.com/c/T2J24OZI/11-confirm-supported-components

Unused blocks are left commented out so they can be easily reintroduced and as a reference.

### CODE REVIEW

**Review by matthewpassmore — CHANGES_REQUESTED** [review]
This constrains the vast amount of blocks, as expected 👍 

We also need to constrain the _Embed_ block variations to just YouTube and Vimeo. Currently there's a whole host of selections e.g. Spotify. You may need a slightly different approach for these. For example, [this is how we did it on NAO](feature/constrain-available-blocks), but there may be a PHP approach that compliments what you've done in this class.

**Comment by sbrody** [comment]
> This constrains the vast amount of blocks, as expected 👍
> 
> We also need to constrain the _Embed_ block variations to just YouTube and Vimeo. Currently there's a whole host of selections e.g. Spotify. You may need a slightly different approach for these. For example, [this is how we did it on NAO](feature/constrain-available-blocks), but there may be a PHP approach that compliments what you've done in this class.

Good point - have done that in JS as it fits with the approach we've taken to block variations

File: app/Blocks/AllowedBlocks.php [hunk 1]
```diff
@@ -13,80 +13,80 @@ public function limitAllowedBlocks($allowed_block_types, $block_editor_context)
...
+				// 'core/rss',
+				// 'core/search',
 				'core/separator',
-				'core/shortcode',
-				'core/site-logo',
-				'core/site-tagline',
-				'core/site-title',
-				'core/social-link',
-				'core/social-links',
+				// 'core/shortcode',
```
Comment (matthewpassmore) [inline app/Blocks/AllowedBlocks.php:80]: This needs to be enabled, to allow continued use of the _Footnotes_ shortcode, if they prefer the existing workflow to the native block.

**Review by matthewpassmore — CHANGES_REQUESTED** [review]

**Review by matthewpassmore — APPROVED** [review]
Looks good to me 👍

**Pull request merged**